src/Axiom/Set.agda

```diff
@@ -298,53 +308,56 @@ record Theoryᵈ : Type₁ where
   open Theory th public
   open Equivalence
 
-  field ∈-sp : ⦃ DecEq A ⦄ → spec-∈ A
-        _∈?_ : ⦃ DecEq A ⦄ → Dec₂ (_∈_ {A = A})
-        all? : ⦃ DecEq A ⦄ → {P : A → Type} (P? : Dec₁ P) {X : Set A} → Dec (All P X)
-        any? : ⦃ DecEq A ⦄ → {P : A → Type} (P? : Dec₁ P) (X : Set A) → Dec (Any P X)
+  field
+    ∈-sp : ⦃ DecEq A ⦄ → spec-∈ A
+    _∈?_ : ⦃ DecEq A ⦄ → Dec₂ (_∈_ {A = A})
+    all? : ⦃ DecEq A ⦄ → {P : A → Type} (P? : Decidable¹ P) {X : Set A} → Dec (All P X)
+    any? : ⦃ DecEq A ⦄ → {P : A → Type} (P? : Decidable¹ P) (X : Set A) → Dec (Any P X)
 
   _∈ᵇ_ : ⦃ DecEq A ⦄ → A → Set A → Bool
```

Comment:
It can't actually, since it takes a visible `A` argument in contrast to the others.

This highlights the more fundamental issue of the decidable part of a `Theory`, which should be defined only when the carrier type has decidable equality, maybe open an issue for that?

Reply:
I'm confused, the module below doesn't bind a hidden `a : A`, only a hidden `A : Set` and instance `DecEq A`, which is exactly what this binds.

But I think you're right - everything in here needs a `DecEq A`. This means we should probably try to refactor this somehow, but if we want `Theoryᵈ` to be similar to `Theory` then we can't add parameters to `Theoryᵈ`. But maybe we don't care about that constraint?

Reply:
Oops, you're right.